docs/website/01_framework/observer.md

```diff
@@ -18,46 +18,64 @@ enum class NonLinearSolverMessages {
+A class can be observable.
+The class then sends notifications when certain events are happening.
+To become observable, a class
+must inherit from `IObservable<MessageType, StateType>`, for example,
 
 ```cpp
-class NewtonRaphson : public IObservable<NonLinearSolverMessages> {...};
+class NewtonRaphson : public IObservable<NonLinearSolverMessages, NonLinearSolverState> {...};
 ```
 
-The function `this->notify(MessageType::Message)` is called at the appropriate position in the code to send a
-notification. This could be, for example,
+Here, ``MessageType``
+is the `enum` of messages to use (see above)
+and ``StateType`` can either be ``NonLinearSolverState`` or ``ControlState``.
+``StateType`` must adhere to the following concept:
 
 ```cpp
-this->notify(NonLinearSolverMessages::SOLUTION_CHANGED);
+template <typename MT, typename ST>
+concept Observable = requires { (std::is_same_v<MT, ControlMessages> && std::is_same_v<ST, ControlState>) ||
+      (std::is_same_v<MT, NonLinearSolverMessages> && std::is_same_v<ST, NonLinearSolverState>); };
+```
+
+The function `this->notify(MessageType message, const StateType& state)` is called at the appropriate position in the code
+to send a notification.
+This could be, for example,
+
+```cpp
+NonLinearSolverState solverState;
+this->notify(NonLinearSolverMessages::SOLUTION_CHANGED, solverState);
 ```
 
 ## IObserver
 
-A class can be an observer. The class is then notified when events are happening and can perform actions. A very simple
-example is shown below. To become an observer, the class must inherit from ``IObserver<MessageType>``, where ``MessageType``
-is the `enum` of messages to use (see above).
+A class can be an observer.
+The class is then notified when events are happening and can perform actions.
+To become an observer, the class must inherit from ``IObserver<IObservable<MessageType, StateType>>``.
+A basic example is shown below.
 
 ```cpp
-class OurFirstObserver : public IObserver<NonLinearSolverMessages> {
+class OurFirstObserver : public IObserver<IObservable<NonLinearSolverMessages, NonLinearSolverState>> {
 public:
-  void updateImpl(NonLinearSolverMessages message) override {
+  void updateImpl(NonLinearSolverMessages message, const Ikarus::NonLinearSolverState &) override {
     if (message == NonLinearSolverMessages::ITERATION_STARTED) std::cout << "Iteration started.\n";
   }
 };
 ```
 
-The observer has to implement the function ``void updateImpl(MessageType message)``. In this function, all actions can
+The observer has to implement the function ``void updateImpl(MessageType message, const StateType& state)``.
+In this function, all actions can
 be implemented that should be performed when the corresponding message is received.
 
-To connect observer and observable, one has to call ``observalbe.subscribe(MessageType::Message,observer)``. Example:
+To connect the observer and the observable, one has to call ``observalbe.subscribe(MessageType::Message, observer)``.
```

Comment:
also typo there. Actually al these descriptions should be removed here and to the doxygen documentation.

Reply:
Also when describing this should be done without the observer object.

Reply:
I fixed the backtick and the typo. Regarding moving these comments to the doxygen documentation, I would propose to do that in a separate PR, where the complete documentation is refactored such that only the framework is discussed in the documentation while such implementation aspects can be moved to doxygen documentation. For example, I think [here](https://ikarus-project.github.io/dev/01_framework/controlRoutines/#path-following-techniques) `SubsidiaryArgs` need not be documented as such because it is better mentioned [here](https://ikarus-project.github.io/dev/doxygen/html/a01221.html). What do you think?

Reply:
I removed such lines